driver/driver.go

```diff
@@ -0,0 +1,633 @@
+	BOND_PRIMARY_RESELECT_BETTER
+	BOND_PRIMARY_RESELECT_FAILURE
+)
+
+func (b BondPrimaryReselect) String() string {
+	switch b {
+	case BOND_PRIMARY_RESELECT_ALWAYS:
+		return "always"
+	case BOND_PRIMARY_RESELECT_BETTER:
+		return "better"
+	case BOND_PRIMARY_RESELECT_FAILURE:
+		return "failure"
+	default:
+		return fmt.Sprintf("unknown BondPrimaryReselect value (%d)", b)
+	}
+}
+
+type BondFailOverMac uint8
+
+const (
+	BOND_FAIL_OVER_MAC_NONE BondFailOverMac = iota
+	BOND_FAIL_OVER_MAC_ACTIVE
+	BOND_FAIL_OVER_MAC_FOLLOW
+)
+
+func (b BondFailOverMac) String() string {
+	switch b {
+	case BOND_FAIL_OVER_MAC_NONE:
+		return "none"
+	case BOND_FAIL_OVER_MAC_ACTIVE:
+		return "active"
+	case BOND_FAIL_OVER_MAC_FOLLOW:
+		return "follow"
+	default:
+		return fmt.Sprintf("unknown BondPrimaryReselect value (%d)", b)
+	}
+}
+
+type BondXmitHashPolicy uint8
+
+const (
+	BOND_XMIT_HASH_POLICY_LAYER2 BondXmitHashPolicy = iota
+	BOND_XMIT_HASH_POLICY_LAYER3_4
+	BOND_XMIT_HASH_POLICY_LAYER2_3
+	BOND_XMIT_HASH_POLICY_ENCAP2_3
+	BOND_XMIT_HASH_POLICY_ENCAP3_4
+	BOND_XMIT_HASH_POLICY_VLAN_SRCMAC
+)
+
+func (b BondXmitHashPolicy) String() string {
+	switch b {
+	case BOND_XMIT_HASH_POLICY_LAYER2:
+		return "layer2"
+	case BOND_XMIT_HASH_POLICY_LAYER3_4:
+		return "layer3+4"
+	case BOND_XMIT_HASH_POLICY_LAYER2_3:
+		return "layer2+3"
+	case BOND_XMIT_HASH_POLICY_ENCAP2_3:
+		return "encap2+3"
+	case BOND_XMIT_HASH_POLICY_ENCAP3_4:
+		return "encap3+4"
+	case BOND_XMIT_HASH_POLICY_VLAN_SRCMAC:
+		return "vlan+srcmac"
+	default:
+		return fmt.Sprintf("unknown BondXmitHashPolicy value (%d)", b)
+	}
+}
+
+type BondAdLacpActive uint8
+
+const (
+	BOND_LACP_ACTIVE_OFF BondAdLacpActive = iota
+	BOND_LACP_ACTIVE_ON
+)
+
+func (b BondAdLacpActive) String() string {
+	switch b {
+	case BOND_LACP_ACTIVE_OFF:
+		return "off"
+	case BOND_LACP_ACTIVE_ON:
+		return "on"
+	default:
+		return fmt.Sprintf("unknown BondLacpActive value (%d)", b)
+	}
+}
+
+type BondLacpRate uint8
+
+const (
+	BOND_LACP_RATE_SLOW BondLacpRate = iota
+	BOND_LACP_RATE_FAST
+)
+
+func (b BondLacpRate) String() string {
+	switch b {
+	case BOND_LACP_RATE_SLOW:
+		return "slow"
+	case BOND_LACP_RATE_FAST:
+		return "fast"
+	default:
+		return fmt.Sprintf("unknown BondLacpRate value (%d)", b)
+	}
+}
+
+type BondAdSelect uint8
+
+const (
+	BOND_AD_SELECT_STABLE BondAdSelect = iota
+	BOND_AD_SELECT_BANDWIDTH
+	BOND_AD_SELECT_COUNT
+)
+
+func (b BondAdSelect) String() string {
+	switch b {
+	case BOND_AD_SELECT_STABLE:
+		return "stable"
+	case BOND_AD_SELECT_BANDWIDTH:
+		return "bandwidth"
+	case BOND_AD_SELECT_COUNT:
+		return "count"
+	default:
+		return fmt.Sprintf("unknown BondAdSelect value (%d)", b)
+	}
+}
+
+type BondAdInfo struct {
+	AggregatorId uint16
+	NumPorts     uint16
+	ActorKey     uint16
+	PartnerKey   uint16
+	PartnerMac   net.HardwareAddr
+}
+
+const bondMaxTargets = 16
+
+// Bond implements LinkDriver for the bond driver
+type Bond struct {
```

Comment:
I understand why all these fields don't have a comment, but would it be possible to link to some kernel documentation that explains these?